platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -64,47 +68,157 @@
+
     return new SolrClientAdapter(
-        core,
+        collection,
         () ->
             AccessController.doPrivileged(
                 (PrivilegedAction<SolrClient>)
                     () -> {
-                      return createSolrCloudClient(zookeeperHosts, core);
+                      return createSolrCloudClient(zookeeperHosts, collection, alias);
                     }));
   }
 
   @VisibleForTesting
-  SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
+  void addCollectionToAlias(String alias, String collection, CloudSolrClient client) {
+    if (StringUtils.isBlank(alias) || StringUtils.isBlank(collection)) {
+      return;
+    }
+    try {
+      List<String> aliasedCollections = getCollectionsInAlias(client, alias);
+      if (!aliasedCollections.contains(collection)) {
+        List<String> collections = new ArrayList<>();
+        collections.addAll(aliasedCollections);
+        collections.add(collection);
+        collections.addAll(getCollectionsNotInAlias(client, aliasedCollections, alias));
+        createAlias(client, collections, alias);
+      }
+    } catch (SolrServerException | IOException e) {
+      LOGGER.warn("Failed to update alias [{}]", alias, e);
+    }
+  }
+
+  private List<String> getCollectionsInAlias(CloudSolrClient client, String alias)
+      throws SolrServerException, IOException {
+    CollectionAdminResponse aliasResponse =
+        new CollectionAdminRequest.ListAliases().process(client);
+
+    if (aliasResponse != null) {
+      Map<String, String> aliases = aliasResponse.getAliases();
+      if (aliases != null && aliases.containsKey(alias)) {
+        String[] collections = aliases.get(alias).split(",");
+        return Arrays.asList(collections);
+      }
+    }
+    return Collections.emptyList();
+  }
+
+  /**
+   * Find existing collections in case parallel operations are creating collections
+   *
+   * @param client
+   * @param currentAliases
+   * @param alias
+   * @return
+   * @throws SolrServerException
+   * @throws IOException
+   */
+  private List<String> getCollectionsNotInAlias(
+      CloudSolrClient client, List<String> currentAliases, String alias)
+      throws SolrServerException, IOException {
+    if (StringUtils.isNotBlank(alias)) {
+      CollectionAdminResponse response = new CollectionAdminRequest.List().process(client);
+      if (response.getResponse() != null) {
+        List<String> collections = (List<String>) response.getResponse().get("collections");
+        return collections == null
+            ? Collections.emptyList()
+            : collections
+                .stream()
+                .filter(c -> c.startsWith(alias))
+                .filter(c -> currentAliases.contains(c))
+                .collect(Collectors.toList());
+      }
+    }
+    return Collections.emptyList();
+  }
+
+  private void createAlias(CloudSolrClient client, List<String> collections, String alias) {
```

Comment:
❓ Per my knowledge of Elasticsearch, an `alias` is just an alternate name you can use to refer to an index or group of indices in a REST request, but the database itself owns the alias definition as part of configuration. It seems Solr works in a similar fashion: 
https://lucene.apache.org/solr/guide/8_1/aliases.html

This is a lot more manageable than before, so I'm good with it 👍 but I'm still curious about the necessity for having our Solr client manage Alias creation. Can we not assume that we deployed Solr / Zookeeper with the correct configuration up front? Can we not simply reboot the system or refresh bundles between configuration updates?

Reply:
Your observation is correct. DDF does bear the responsibly to setup solr if it is not configured. But if Solr is already configured, either manually or on previous DDF run, then the current or update configuration won't have any impact.